### ISSUE TITLE
The configs views.view.node_results.json and views.view.user_results.json are just copies to each other and make conflict.

### ISSUE THREAD

**Issue by paucku** [backdrop-contrib/h5p#8]
The configs `views.view.node_results.json` and `views.view.user_results.json` are just copies to each other.

This make conlfict and prevents the module to uninstal completely – it leaves some files in .../config/active folder which cannot be removed.

I think that inside `views.view.user_results.json` everything which is node related should be renamed as user related.

**Comment by argiepiano** [comment]
No, those files are not removed because they are Views configurations, and Views owns them. As long as Views is installed (which is a required module in BAckdrop), those Views will remain after you uninstall other modules that created them. To remove Views json files, modules must explicitly remove them in hook_uninstall().

**Comment by laryn** [comment]
I've not used this module, but I see what you are referring to @paucku -- the user view appears to be a copy of the node view.

@argiepiano -- if a module adds a `module` key in the Views config to indicate ownership, do you know if that is enough to make those Views uninstall automatically with the module? I vaguely thought so but could be wrong.

**Comment by laryn** [comment]
Answering my own question (kind of), here's a commit from a project that I was thinking of which adds the `module` key -- but it also disables/uninstalls views in the associated hooks:

 - https://github.com/backdrop-contrib/ubercart/commit/2ca99b6565aa997d027a89359a4cae7a5405d21f

**Comment by paucku** [comment]
My main point (which I did not clarify enough) is that the two configs in the two JSON files **have the same name** – node_results. That makes a conflict.

I _do not_ refer to that the two views are not removed autmatically on uninstall (which I don't know is or isn't a desirable behavior).

However I do say that you **cannot remove them even mannually**. 

Steps to reproduce:

1. Install the module h5p
2. Enable the modules H5P and H5P Editor
3. Go to Views – there are two views **with the same machine name** 
<img width="1158" alt="Screenshot 2024-08-27 at 23 58 32" src="https://github.com/user-attachments/assets/a2d5d28f-97e9-449d-af48-0b0910b8f746">

4. Disable the modules H5P and H5P Editor
5. Uninstall the modules H5P Editor and H5P 
6. The two views with the same machine names are still there
7. Delete one of them
8. You may receive an error but maybe not
9. Delete the other view
10. You cannot – it tries to delete the view which we already deleted because it has the same machine name – node_results.

Now you have a permanent undeletable view wich is broken. If you try to delete it it gets you to nowehere (because we've already deleted the with machine name node_result):
<img width="833" alt="Screenshot 2024-08-28 at 0 11 48" src="https://github.com/user-attachments/assets/abab8904-424b-4ab7-b976-40c1f40d65c3">

So at least the machine names of the two views should be different.

**Comment by argiepiano** [comment]
Ah, I see what you mean. Yes, there is a line that's wrong in views.view.user_results.json

The line: 
  "name": "node_results",

Should be
"name": "user_results",

**Comment by paucku** [comment]
@argiepiano, what should I do now? Should I clone it repair it and make a pull request?

**Comment by argiepiano** [comment]
@paucku, the issue you discovered is more serious than what I said above. The View stored in `modules/h5p/config/views.view.user_results.json` is actually not a User view, but rather (as you discovered) a clone of the node view with a different name. And as I mentioned above, the view machine name doesn't match the file name, which is also a problem. 

I'm not familiar with the module at all, so I don't know what these two Views are supposed to do. But my guess is that, if you want to provide a PR, this is what I may do:

1. Fork the repo and clone it locally into your installation
2. Remove `modules/h5p/config/views.view.user_results.json` before enabling the module for the first time, so that this faulty view doesn't get installed
3. Enable the module
4. Create a User view from scratch (again, I don't know what the goal of this user View should be, or what fields it should have)
5. Save the new view
6. Copy the new View file you just created from your active config folder, typically in `files/config_XXX/active/` into the module's `config` folder

Then submit the PR from this.

Again, I don't know what this view is supposed to do, so trying to fix the existing View may not work at all. That's why I 'd build it from scratch...

**Comment by paucku** [comment]
@argiepiano, thank you for the instruciton. I have idea what this module and view are supposed to do so I will try and test some things.
